test/performance/tests/test_util/harness/lxd.py

```diff
@@ -160,8 +161,35 @@ def exec(self, instance_id: str, command: list, **kwargs):
             raise HarnessError(f"unknown instance {instance_id}")
 
         LOG.debug("Execute command %s in instance %s", command, instance_id)
+        # Check if the command contains any special shell characters that need special handling
+        special_characters = ["$", ">", "&", "!", "~", "||", "&&", "|"]
```

Comment:
Do we care about other characters as well? e.g. `<`
Also I think `&&` and `||` can be removed, we have single versions of them (AFAIK `char in " ".join(command)` will catch those).

Reply:
A little more information here, I've added this if clause as `shlex.join` did not work with any of my commands with special characters.

So now this line checks for any of those special characters and does a regular join on the command bits:
```
any(char in " ".join(command) for char in special_characters)
```

Reply:
Gotcha. Thanks!

Reply:
I removed some duplication and reduced special char to the one I need :)